treeherder/etl/management/commands/ingest.py

```diff
@@ -149,10 +219,16 @@ async def await_futures(fs):
 
 
 def process_job_with_threads(pulse_job, root_url):
-    acquire_connection()
     logger.info("Loading into DB:\t%s", pulse_job["taskId"])
-    JobLoader().process_job(pulse_job, root_url)
-    release_connection()
+    try:
+        acquire_connection()
```

Comment:
replace 

```python

def acquire_connection():
    """
    Decrement database resource count. If resource count is 0, block thread
    until resource count becomes 1 before decrementing again.
    """
    conn_sem.acquire()


def release_connection():
    """
    Close thread's conneciton to database and increment database resource count.
    """
    connection.close()
    conn_sem.release()

```

with  

```python
class Connection(object):

    def __enter__(self):
        conn_sem.acquire()
        
    def __exit__(self, exc_type, exc_val, exc_tb):
        connection.close()
        conn_sem.release()
```

then you may 

```python

with Connection():
    try:
        JobLoader().process_job(pulse_job, root_url)
    except MissingPushException:
        logger.warning('The push was not in the DB. We are going to try that first')
        _ingest_push(pulse_job["origin"]["project"], pulse_job["origin"]["revision"])
        JobLoader().process_job(pulse_job, root_url)

Reply:
... one other thing, 

```python
        connection.close()
        conn_sem.release()
```

I am a bit concerned that the connection is closed, but apparently never opened.  I assume this works despite the lack of symmetry.